File: web-app/src/lib/features/features/project.ts
import type { Feature } from '../types';

export const projectFeatures = {
  'project-create': {
    id: 'project-create',
    title: 'Create Project',
    description: 'Creates a new project for grouping and organizing sessions.',
    rpcIds: ['project:create'],
    componentPaths: [],
    testIds: [
      'TestCreateProject_Success',
      'TestCreateProject_EmptyName',
      'TestCreateProject_NilStorage',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'project-list': {
    id: 'project-list',
    title: 'List Projects',
    description: 'Lists all projects available in the current workspace.',
    rpcIds: ['project:list'],
    componentPaths: [],
    testIds: [
      'TestListProjects_EmptyInitially',
      'TestListProjects_AfterCreate',
      'TestListProjects_NilStorageReturnsEmpty',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'project-update': {
    id: 'project-update',
    title: 'Update Project',
    description: 'Updates the name or metadata of an existing project.',
    rpcIds: ['project:update'],
    componentPaths: [],
    testIds: [
      'TestUpdateProject_Success',
      'TestUpdateProject_NilStorage',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'project-delete': {
    id: 'project-delete',
    title: 'Delete Project',
    description: 'Deletes a project and disassociates its sessions.',
    rpcIds: ['project:delete'],
    componentPaths: [],
    testIds: [
      'TestDeleteProject_Success',
      'TestDeleteProject_EmptyID',
      'TestDeleteProject_NilStorage',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'project-assign-sessions': {
    id: 'project-assign-sessions',
    title: 'Assign Sessions To Project',
    description: 'Assigns one or more sessions to a project for organizational grouping.',
    rpcIds: ['project:assign-sessions'],
    componentPaths: [],
    testIds: [
      'TestAssignSessionsToProject_EmptyProjectID',
      'TestAssignSessionsToProject_NilStorage',
      'TestAssignSessionsToProject_EmptySessions',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/project.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// ProjectCreate describes the create project RPC.
var ProjectCreate = featureregistry.Feature{
	ID:          "project-create",
	Title:       "Create Project",
	Description: "Creates a new project for grouping and organizing sessions.",
	RPCIDs:      []string{"project:create"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// ProjectList describes the list projects RPC.
var ProjectList = featureregistry.Feature{
	ID:          "project-list",
	Title:       "List Projects",
	Description: "Lists all projects available in the current workspace.",
	RPCIDs:      []string{"project:list"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// ProjectUpdate describes the update project RPC.
var ProjectUpdate = featureregistry.Feature{
	ID:          "project-update",
	Title:       "Update Project",
	Description: "Updates the name or metadata of an existing project.",
	RPCIDs:      []string{"project:update"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// ProjectDelete describes the delete project RPC.
var ProjectDelete = featureregistry.Feature{
	ID:          "project-delete",
	Title:       "Delete Project",
	Description: "Deletes a project and disassociates its sessions.",
	RPCIDs:      []string{"project:delete"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// ProjectAssignSessions describes the assign sessions to project RPC.
var ProjectAssignSessions = featureregistry.Feature{
	ID:          "project-assign-sessions",
	Title:       "Assign Sessions To Project",
	Description: "Assigns one or more sessions to a project for organizational grouping.",
	RPCIDs:      []string{"project:assign-sessions"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(ProjectCreate)
	featureregistry.Register(ProjectList)
	featureregistry.Register(ProjectUpdate)
	featureregistry.Register(ProjectDelete)
	featureregistry.Register(ProjectAssignSessions)
}
